tests/components/insteon/test_config_flow.py

```diff
@@ -0,0 +1,703 @@
+    )
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "cannot_connect"}
+
+
+async def test_failed_connection_hub(hass: HomeAssistantType):
+    """Test a failed connection with a Hub."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await _init_form(hass, HUB2)
+
+    result2, _, _ = await _device_form(
+        hass, result["flow_id"], mock_failed_connection, MOCK_USER_INPUT_HUB_V2
+    )
+    assert result2["type"] == "form"
+    assert result2["errors"] == {"base": "cannot_connect"}
+
+
+async def _import_config(hass, config):
+    """Run the import step."""
+    with patch(PATCH_CONNECTION, new=mock_successful_connection,), patch(
+        PATCH_ASYNC_SETUP, return_value=True
+    ), patch(PATCH_ASYNC_SETUP_ENTRY, return_value=True):
+        return await async_import_config(hass, config)
+
+
+async def test_import_plm(hass: HomeAssistantType):
+    """Test importing a minimum PLM config from yaml."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    result = await _import_config(hass, MOCK_IMPORT_CONFIG_PLM)
+
+    assert result["type"] == "create_entry"
+    assert hass.config_entries.async_entries(DOMAIN)
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        assert entry.data == MOCK_USER_INPUT_PLM
+
+
+async def test_import_plm_full(hass: HomeAssistantType):
+    """Test importing a full PLM config from yaml."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+    result = await _import_config(hass, MOCK_IMPORT_FULL_CONFIG_PLM)
+    assert result["type"] == "create_entry"
+    assert hass.config_entries.async_entries(DOMAIN)
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        assert entry.data == MOCK_USER_INPUT_PLM
+        assert entry.options[CONF_OVERRIDE][0][CONF_ADDRESS] == "1A.2B.3C"
+        assert entry.options[CONF_OVERRIDE][0][CONF_CAT] == MOCK_CAT
+        assert entry.options[CONF_OVERRIDE][0][CONF_SUBCAT] == MOCK_SUBCAT
+        assert entry.options[CONF_X10][0][CONF_HOUSECODE] == MOCK_HOUSECODE
+        assert entry.options[CONF_X10][0][CONF_UNITCODE] == MOCK_UNITCODE
+        assert entry.options[CONF_X10][0][CONF_PLATFORM] == MOCK_X10_PLATFORM
+        assert entry.options[CONF_X10][0][CONF_DIM_STEPS] == MOCK_X10_STEPS
+
+
+async def test_import_full_hub_v1(hass: HomeAssistantType):
+    """Test importing a full Hub v1 config from yaml."""
+    await setup.async_setup_component(hass, "persistent_notification", {})
+
+    result = await _import_config(hass, MOCK_IMPORT_FULL_CONFIG_HUB_V1)
+
+    assert result["type"] == "create_entry"
+    assert hass.config_entries.async_entries(DOMAIN)
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        assert entry.data[CONF_HOST] == MOCK_HOSTNAME
+        assert entry.data[CONF_PORT] == MOCK_PORT
+        assert entry.data[CONF_HUB_VERSION] == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)